spec.html

```diff
@@ -11822,10 +11834,10 @@ <h1>GetGlobalObject ( ): an Object</h1>
   <emu-clause id="sec-jobs" oldids="sec-jobs-and-job-queues,sec-enqueuejob,sec-runjobs,job-queue">
     <h1>Jobs and Host Operations to Enqueue Jobs</h1>
     <p>A <dfn id="job" variants="Jobs">Job</dfn> is an Abstract Closure with no parameters that initiates an ECMAScript computation when no other ECMAScript computation is currently in progress.</p>
-    <p>Jobs are scheduled for execution by ECMAScript host environments. This specification describes the host hook HostEnqueuePromiseJob to schedule one kind of job; hosts may define additional abstract operations which schedule jobs. Such operations accept a Job Abstract Closure as the parameter and schedule it to be performed at some future time. Their implementations must conform to the following requirements:</p>
+    <p>Jobs are scheduled for execution by ECMAScript host environments in an agent. A Job is usually, but not always, executed in the same agent that enqueued it. This specification describes the host hooks HostEnqueueGenericJob, HostEnqueueFinalizationRegistryCleanupJob, HostEnqueuePromiseJob, and HostEnqueueTimeoutJob to schedule jobs. The host hooks in this specification are organized by the additional constraints imposed on the scheduling of jobs. Hosts may define additional abstract operations which schedule jobs. Such operations accept a Job Abstract Closure as the parameter and schedule it to be performed at some future time. Their implementations must conform to the following requirements:</p>
 
     <ul>
-      <li>At some future point in time, when there is no running execution context and the execution context stack is empty, the implementation must:
+      <li>At some future point in time in some agent _agent_, when there is no running execution context in _agent_ and its execution context stack is empty, the implementation must:
```

Comment:
This sounds a bit like we're saying it's generally up to the host which agent to run the job in. That's not what we want, I think - it's always going to be specified.

I think instead

```suggestion
      <li>At some future point in time, when there is no running execution context in the agent for which the job is enqueued and that agent's execution context stack is empty, the implementation must:
```

And then also tweak the paragraph above to make it clear that jobs are scheduled in particular agents:

```
Jobs are scheduled for execution by ECMAScript host environments in a particular agent
```

Reply:
I think given https://github.com/whatwg/html/pull/4613#discussion_r1193326552, we should probably make all host hooks pass an agent / global.

Reply:
I've changed the language to say jobs are scheduled to run in a particular realm in a particular agent, and that the hooks take a job and a realm as parameters. HostEnqueueFinalizationRegistryCleanupJob doesn't follow this strictly speaking right now anyway (because it doesn't take a job as an argument, but a `FinalizationRegistry`), which I will fix in a follow up PR.